File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    
    var answer = 0;
    
    for(let i=0; i<s.length; i++){
        
        const que = [];
        
        for(let j=0; j<s.length; j++){
            if(!que.length){
                que.push(s.charAt(j));
            }else if(s.charAt(j) == '}' && que[que.length-1] == '{'){
                que.pop();
            }else if(s.charAt(j) == ']' && que[que.length-1] == '['){
                que.pop();
            }else if(s.charAt(j) == ')' && que[que.length-1] == '('){
                que.pop();
            }else{
                que.push(s.charAt(j));
            }
        }
        
        if(!que.length) {
            answer++;
        }
        
        s = s.substring(1) + s.substring(0, 1);
    }

    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        
        int answer = 0;
        
        for(int i=0; i<s.length(); i++){
            Deque<Character> que = new ArrayDeque<>();
            for(int j=0; j<s.length(); j++){
                if(que.isEmpty()){
                    que.add(s.charAt(j));
                }else if(s.charAt(j) == '}' && que.peekLast() == '{'){
                    que.pollLast();
                }else if(s.charAt(j) == ']' && que.peekLast() == '['){
                    que.pollLast();
                }else if(s.charAt(j) == ')' && que.peekLast() == '('){
                    que.pollLast();
                }else{
                    que.add(s.charAt(j));
                }
            }
            if(que.isEmpty()) answer++;
            s = s.substring(1) + s.substring(0, 1);
        }
        
        return answer;
    }
}